frontend/pages/SoftwarePage/SoftwareTitleDetailsPage/ViewYamlModal/helpers.tsx

```diff
@@ -0,0 +1,181 @@
+      ),
+    });
+  }
+  if (uninstallScript) {
+    items.push({
+      key: "uninstall-script",
+      element: (
+        <Button
+          key="uninstall"
+          variant="text-link"
+          onClick={onClickUninstallScript}
+        >
+          uninstall script
+        </Button>
+      ),
+    });
+  }
+  if (postInstallScript) {
+    items.push({
+      key: "post-install-script",
+      element: (
+        <Button
+          key="post"
+          variant="text-link"
+          onClick={onClickPostInstallScript}
+        >
+          post-install script
+        </Button>
+      ),
+    });
+  }
+
+  if (items.length === 0) return <></>;
+
+  // Helper to join items with commas and Oxford comma before "and"
+  const joinWithCommasAnd = (
+    elements: { key: string; element: JSX.Element }[]
+  ) => {
+    return elements.map((item, idx) => {
+      if (idx === 0) return item.element;
+      if (idx === elements.length - 1) {
+        return (
+          <React.Fragment key={`and-${item.key}`}>
+            {elements.length > 2 ? "," : ""} and {item.element}
+          </React.Fragment>
+        );
+      }
+      return (
+        <React.Fragment key={`comma-${item.key}`}>
+          , {item.element}
+        </React.Fragment>
+      );
+    });
+  };
+
+  return (
+    <>
+      Next, download your {joinWithCommasAnd(items)} and add{" "}
+      {items.length === 1 ? "it" : "them"} to your repository (please use the
+      above {items.length === 1 ? "path" : "paths"}).
+    </>
+  );
+};
+
+/** Hyphenate the name for file paths */
+export const hyphenatedSoftwareTitle = (softwareTitle: string): string => {
+  return softwareTitle.trim().toLowerCase().replace(/\s+/g, "-");
+};
+
+interface CreatePackageYamlParams {
+  softwareTitle: string;
+  packageName: string;
+  version: string;
+  url?: string;
+  sha256?: string | null;
+  preInstallQuery?: string;
+  installScript?: string;
+  postInstallScript?: string;
+  uninstallScript?: string;
+}
+
+export const createPackageYaml = ({
+  softwareTitle,
+  packageName,
+  version,
+  url,
+  sha256,
+  preInstallQuery,
+  installScript,
+  postInstallScript,
+  uninstallScript,
+}: CreatePackageYamlParams): string => {
+  let yaml = `# ${softwareTitle} (${packageName}) version ${version}
+`;
+
+  if (url) {
+    yaml += `url: ${url}
+`;
+  }
+
+  if (sha256) {
+    yaml += `hash_sha256: ${sha256}
+`;
+  }
+
+  const hyphenatedSWTitle = hyphenatedSoftwareTitle(softwareTitle);
+
+  if (preInstallQuery) {
+    yaml += `pre_install_query:
+  path: ../queries/pre-install-query-${hyphenatedSWTitle}.yml
+`;
+  }
+
+  if (installScript) {
+    yaml += `install_script:
+  path: ../scripts/install-${hyphenatedSWTitle}.sh
+`;
+  }
+
+  if (postInstallScript) {
+    yaml += `post_install_script:
+  path: ../scripts/post-install-${hyphenatedSWTitle}.sh
+`;
+  }
+
+  if (uninstallScript) {
+    yaml += `uninstall_script:
+  path: ../scripts/uninstall-${hyphenatedSWTitle}.sh
+`;
+  }
```

Comment:
FYI the `generate-gitops` command exports these into team-specific folders e.g. `../scripts/workstations` and exports the files as e.g. `<software name>-<platform>-install.sh`.  We should aim for consistency here IMO.  I think the <platform> part is important regardless since we can have both Windows and MacOS versions of 1password, for example.

Reply:
@eugkuo @mostlikelee ---

Ah yes, circling back to our naming conversation Monday. Looks like we overlooked `generate-gitops` current commands and should be consistent!

Let's confirm during DR today.